From: Robin Dong <sanbai@taobao.com>
Subject: [PATCH 04/09] block: export symbol __make_request()
Patch-mainline:

It need __make_request to make request for flashcache.

Signed-off-by: Robin Dong <sanbai@taobao.com>
---
Index: linux-2.6.32-279.9.1.el5/block/blk-core.c
===================================================================
--- linux-2.6.32-279.9.1.el5.orig/block/blk-core.c
+++ linux-2.6.32-279.9.1.el5/block/blk-core.c
@@ -39,8 +39,6 @@ EXPORT_TRACEPOINT_SYMBOL_GPL(block_remap
 EXPORT_TRACEPOINT_SYMBOL_GPL(block_rq_remap);
 EXPORT_TRACEPOINT_SYMBOL_GPL(block_bio_complete);
 
-static int __make_request(struct request_queue *q, struct bio *bio);
-
 /*
  * For the allocated request tables
  */
@@ -1362,7 +1360,7 @@ static void blk_account_io_front_merge(s
 	}
 }
 
-static int __make_request(struct request_queue *q, struct bio *bio)
+int __make_request(struct request_queue *q, struct bio *bio)
 {
 	struct request *req;
 	int el_ret;
@@ -1513,6 +1511,7 @@ out_unlock:
 	spin_unlock_irq(q->queue_lock);
 	return 0;
 }
+EXPORT_SYMBOL_GPL(__make_request);
 
 /*
  * If bio->bi_dev is a partition, remap the location
Index: linux-2.6.32-279.9.1.el5/include/linux/blkdev.h
===================================================================
--- linux-2.6.32-279.9.1.el5.orig/include/linux/blkdev.h
+++ linux-2.6.32-279.9.1.el5/include/linux/blkdev.h
@@ -770,6 +770,7 @@ extern int scsi_cmd_ioctl(struct request
 			  unsigned int, void __user *);
 extern int sg_scsi_ioctl(struct request_queue *, struct gendisk *, fmode_t,
 			 struct scsi_ioctl_command __user *);
+extern int __make_request(struct request_queue *q, struct bio *bio);
 
 /*
  * A queue has just exitted congestion.  Note this in the global counter of
