From: Li Yu <bingtian.ly@taobao.com>
Subject: rt-stat: skip the TCP connection that SYN-ACK is ever retransmitted.
Patch-mainline: In house
References: 

Skip these connections since they prolong measured latency wrongly.

Acked-by: 
Signed-off-by: Li Yu <bingtian.ly@taobao.com>

Index: linux-2.6.32-279.2.1.el5/net/ipv4/rt_stat.c
===================================================================
--- linux-2.6.32-279.2.1.el5.orig/net/ipv4/rt_stat.c
+++ linux-2.6.32-279.2.1.el5/net/ipv4/rt_stat.c
@@ -376,7 +376,9 @@ static inline struct sock *do_tcp_rt_sta
 
 	preempt_disable();
 	cpu = smp_processor_id();
-	if (req->ts_incoming && AF_INET == req->rsk_ops->family) {
+	if (req->ts_incoming
+			&& !req->retrans
+			&& AF_INET == req->rsk_ops->family) {
 		tcp_rt_stat(cpu, req);
 		req->ts_incoming = 0;
 	}
